Sources/MessagingInApp/Config/MessagingInAppConfigBuilder.swift

```diff
@@ -39,26 +39,51 @@ public class MessagingInAppConfigBuilder {
 
 /// Defines errors that can occur during the `MessagingInAppConfigBuilder` building process.
 public enum MessagingInAppConfigBuilderError: Error {
+    case malformedConfig
     case missingSiteId
-    case missingRegion
+    case invalidRegionType
 }
 
 public extension MessagingInAppConfigBuilder {
     /// Constants used to map each of the options in MessagingInAppConfigOptions.
-    enum Keys: String {
+    private enum Keys: String {
         case siteId
         case region
     }
 
     /// Constructs `MessagingInAppConfigOptions` by parsing and applying configurations from provided dictionary.
+    /// The method is used to create `MessagingInAppConfigOptions` from wrapper SDK configurations.
     @available(iOSApplicationExtension, unavailable)
-    static func build(from dictionary: [String: Any]) throws -> MessagingInAppConfigOptions {
-        guard let siteId = dictionary[Keys.siteId.rawValue] as? String else {
+    static func build(from sdkConfig: [String: Any?], withConfig inAppConfigKey: String = "inApp") throws -> MessagingInAppConfigOptions? {
```

Comment:
do we need `inAppConfigKey` as parameter since it's in the `MessagingInApp` module already? and we have a defined key for it?

Reply:
Yeah, looks like an overkill unless we really need to handle different keys for different wrapper SDKs. Removing it for now.